sdk/src/client/api/wait_for_tx_acceptance.rs

```diff
@@ -25,17 +25,31 @@ impl Client {
             .map(std::time::Duration::from_millis)
             .unwrap_or(DEFAULT_WAIT_FOR_TX_ACCEPTANCE_INTERVAL);
 
-        for _ in 0..max_attempts.unwrap_or(DEFAULT_WAIT_FOR_TX_ACCEPTANCE_MAX_ATTEMPTS) {
-            let transaction_metadata = self.get_transaction_metadata(transaction_id).await?;
-
-            match transaction_metadata.transaction_state {
-                TransactionState::Accepted | TransactionState::Committed | TransactionState::Finalized => {
-                    return Ok(());
+        for attempt in 0..max_attempts.unwrap_or(DEFAULT_WAIT_FOR_TX_ACCEPTANCE_MAX_ATTEMPTS) {
+            // Ignore not found for the first few attempts, as the node may not have fully processed the transaction
+            // yet.
+            let transaction_metadata = if attempt < 3 {
```

Comment:
This seems...bad. Is there no way for us to do this better?

Reply:
> Well, the node should always be able to find the transaction, right?

Not if it didn't get the transaction yet (because it's not the node you submitted the block to and the nodes didn't gossip it to this node yet) or haven't fully processed it which is exactly what happened here as you can see in the debug logs of the PR description. First request it didn't return it, ~half a second later it did (the transaction had > 100 outputs)

Reply:
Yeah but I mean it will eventually find it, always, assuming you sent it

Reply:
But that doesn't help if we return an error because it wasn't found the first attempt

Reply:
Nvm just understood you comment now :joy: Yes I guess we can also just always ignore not found errors

Reply:
Wouldn't even hang forever as the attempts are limited

Reply:
Updated